src/atomic.rs

```diff
@@ -328,10 +335,22 @@ impl AtomicBlock {
 
 impl Debug for AtomicBlock {
     fn fmt(&self, f: &mut Formatter) -> Result<(), FormatError> {
-        f.debug_struct("AtomicBlock")
-            .field("mask", &self.mask)
-            .field("atom", &self.atom.get().unwrap().iter())
-            .finish()
+        let mut s = f.debug_struct("AtomicBlock");
+        s.field("mask", &self.mask);
+        if let Some(atom) = self.atom.get() {
+            s.field(
+                "atom",
+                &atom
```

Comment:
if this returns `None` I think we would still want the formatting to show that an `atom` field exists

e.g. in the `None` case perhaps it could be:
```rs
s.field("atom", &"None");
```
or following from my other comment to use a helper type we could do:
```rs
s.field("atom", &self.atom.get().map(DebugAtom));
```